go/service/saltpack.go

```diff
@@ -359,11 +359,13 @@ func (h *SaltpackHandler) saltpackEncryptFile(ctx context.Context, arg keybase1.
 	}, nil
 }
 
-func (h *SaltpackHandler) saltpackEncryptDirectory(ctx context.Context, arg keybase1.SaltpackEncryptFileArg) (keybase1.SaltpackEncryptFileResult, error) {
+func (h *SaltpackHandler) saltpackEncryptDirectory(ctx context.Context, arg keybase1.SaltpackEncryptFileArg) (res keybase1.SaltpackEncryptFileResult, err error) {
+	defer h.G().NotifyRouter.HandleSaltpackOperationDone(ctx, keybase1.SaltpackOperationType_ENCRYPT, arg.Filename)
```

Comment:
I would put this up a level at all the handlers that are in the "frontend interface" section of saltpack.avdl.  i.e. encrypting a single file could have a progress bar that would need `done` to be called if there is an error.

Reply:
the encrypt/sign file methods are already always calling saltpackstart and saltpackdone becaues they use the `sourceFile` struct and its methods.

so i think all we need to do is make sure that encrypt/sign directory always call (defers) saltpackdone if they call saltpackstart.

Reply:
👍